ethpm/utils/cache.py

```diff
@@ -0,0 +1,27 @@
+from typing import Any, Callable
+
+
+class cached_property:
+    """
+    Decorator that converts a method with a single self argument into a
+    property cached on the instance.
+
+    Optional ``name`` argument allows you to make cached properties of other
+    methods. (e.g.  url = cached_property(get_absolute_url, name='url') )
+    """
+
+    def __init__(self, func: Callable[[Any], Any], name: str = None) -> None:
+        self.func = func
+        self.__doc__ = getattr(func, "__doc__")
```

Comment:
It might be better to use [`functools.update_wrapper`](https://docs.python.org/3.5/library/functools.html#functools.update_wrapper) to do this.  It'll move all of `__name__`, `__module__`, `__qualname__`, `__doc__` and `__annotactions__` as well as mergind `__dict__` on the two.  I think all of these are good things.